src/internal/m365/service/exchange/export_test.go

```diff
@@ -0,0 +1,275 @@
+						},
+						&dataMock.Item{
+							ItemID: "id2",
+							Reader: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+						},
+					},
+				},
+			},
+			expectedItems: []export.Item{
+				{
+					ID:   "id1",
+					Name: "id1.eml",
+					Body: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+				},
+				{
+					ID:   "id2",
+					Name: "id2.eml",
+					Body: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+				},
+			},
+		},
+		{
+			name:    "items with success and fetch error",
+			version: version.Groups9Update,
+			backingCollection: data.FetchRestoreCollection{
+				Collection: dataMock.Collection{
+					ItemData: []data.Item{
+						&dataMock.Item{
+							ItemID: "id0",
+							Reader: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+						},
+						&dataMock.Item{
+							ItemID:  "id1",
+							ReadErr: assert.AnError,
+						},
+						&dataMock.Item{
+							ItemID: "id2",
+							Reader: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+						},
+					},
+				},
+			},
+			expectedItems: []export.Item{
+				{
+					ID:   "id0",
+					Name: "id0.eml",
+					Body: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+				},
+				{
+					ID:   "id2",
+					Name: "id2.eml",
+					Body: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+				},
+				{
+					ID:    "",
+					Error: assert.AnError,
+				},
+			},
+		},
+	}
+
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			t := suite.T()
+
+			ctx, flush := tester.NewContext(t)
+			defer flush()
+
+			stats := data.ExportStats{}
+			ec := exchange.NewExportCollection(
+				"",
+				[]data.RestoreCollection{test.backingCollection},
+				test.version,
+				&stats)
+
+			items := ec.Items(ctx)
+
+			count := 0
+			size := 0
+			fitems := []export.Item{}
+
+			for item := range items {
+				if item.Error == nil {
+					count++
+				}
+
+				if item.Body != nil {
+					b, err := io.ReadAll(item.Body)
+					assert.NoError(t, err, clues.ToCore(err))
+
+					size += len(b)
+					item.Body = io.NopCloser(bytes.NewBuffer(b))
+				}
+
+				fitems = append(fitems, item)
+			}
+
+			assert.Len(t, fitems, len(test.expectedItems), "num of items")
+
+			// We do not have any grantees about the ordering of the
+			// items in the SDK, but leaving the test this way for now
+			// to simplify testing.
+			for i, item := range fitems {
+				assert.Equal(t, test.expectedItems[i].ID, item.ID, "id")
+				assert.Equal(t, test.expectedItems[i].Name, item.Name, "name")
+				assert.ErrorIs(t, item.Error, test.expectedItems[i].Error)
+			}
+
+			var expectedStats data.ExportStats
+
+			if size+count > 0 { // it is only initialized if we have something
+				expectedStats = data.ExportStats{}
+				expectedStats.UpdateBytes(path.EmailCategory, int64(size))
+
+				for i := 0; i < count; i++ {
+					expectedStats.UpdateResourceCount(path.EmailCategory)
+				}
+			}
+
+			assert.Equal(t, expectedStats, stats, "stats")
+		})
+	}
+}
+
+func (suite *ExportUnitSuite) TestExportRestoreCollections() {
```

Comment:
could you please add a few more test cases (or maybe just expand this test?) that ensure we properly handle multiple collections, some of which may not have the data category that we expect